app/homepage/index.js

```diff
@@ -73,13 +73,53 @@ const getUsername = () => {
 		.then(({ username }) => {
 			document.getElementById(
 				'logged-in'
-			).textContent = `${username}, welcome to NASA`;
+			).textContent = `Welcome to your favourite astronomy page, ${username}!`;
+			document.getElementById('login-link').remove();
+			document.getElementById('register-link').remove();
+
+			const roverMapList = document.getElementById('navbar');
+
+			const roverMapLink = document.createElement('a');
+			roverMapLink.href = '/user/rover';
+			roverMapLink.textContent = 'Rover';
+			roverMapList.appendChild(roverMapLink);
+
+			const calculatorLink = document.createElement('a');
+			calculatorLink.href = '/user/calculator';
+			calculatorLink.textContent = 'Calculator';
+			roverMapList.appendChild(calculatorLink);
+
+			const userAccountLink = document.createElement('a');
+			userAccountLink.href = '/user/account';
+			userAccountLink.textContent = 'Account';
+			roverMapList.appendChild(userAccountLink);
+
+			const logoutLink = document.createElement('a');
+			logoutLink.href = '/user/logout';
+			logoutLink.textContent = 'Logout';
+			roverMapList.appendChild(logoutLink);
```

Comment:
Interesting! I like the navbar that changes links based on whether or not the user's logged in!

My first thought was that we should see if we can't DRY this bit out.

My second thought is to do with separation between HTML and JS: let's put our structural elements in the HTML — i.e. start with all the links in the HTML — and then apply customised CSS classes to them that depend on the user's logged-in status — i.e. update their CSS classes in the JS.

Reply:
Discussed and resolved.